src/util/message.cpp

```diff
@@ -72,12 +77,27 @@ bool MessageSign(
     return true;
 }
 
-uint256 MessageHash(const std::string& message)
+uint256 MessageHash(const std::string& message, MessageSignatureFormat format)
 {
-    HashWriter hasher{};
-    hasher << MESSAGE_MAGIC << message;
-
-    return hasher.GetHash();
+    switch (format) {
+    case MessageSignatureFormat::LEGACY:
+        {
+            HashWriter hasher{};
+            hasher << MESSAGE_MAGIC << message;
+            return hasher.GetHash();
+        }
+
+    case MessageSignatureFormat::SIMPLE:
+    case MessageSignatureFormat::FULL:
+        {
+            HashWriter hasher{HASHER_BIP322};
+            if (!message.empty()) {
+                hasher.write(AsBytes(Span{message.data(), message.size() * sizeof(char)}));
```

Comment:
nit: Considering that `sizeof(char)` is always 1 (see e.g. https://www.cs.technion.ac.il/users/yechiel/c++-faq/sizeof-char.html), this multiplication is not needed:
```suggestion
                hasher.write(AsBytes(Span{message.data(), message.size()}));
```